### PR TITLE
update documentation

### PR DESCRIPTION
**Pull request by Qwertic**
### Description 
 Adding new endpoints 
- Auth API
- Subscriptions API

Adding new object
- Webhook

Minor fixes on the docs

### CODE REVIEW

File: docs/api_reference/auth_api/introduction.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+sidebar_position: 1
+---
+
+# Introduction
+
+Auth API allows to authenticate from external APIs.
```
Comment (proudlygeek) [inline docs/api_reference/auth_api/introduction.md]: This is not 100% correct, right? Since authorization is implicit in the token. Maybe this should be `verify that an API token is valid`?

File: docs/api_reference/contacts_api/get_contacts.md [hunk 2]
```diff
@@ -15,11 +15,10 @@ List all contacts belonging to the account. A filter can be specified in order t
 | `source`  | Source   | The integration type (e.g. `whatsapp`)                                             |
 | `tags`    | string[] | The matching tags, comma-separated (e.g. `sales,lead`). Tags are _case-insentive_. |
 
-
 ### Example Request
 
 ```bash title=request.sh
-curl -X GET "https://api.callbell.eu/v1/contacts?page=1&source=whatsapp&tags=sales,lead" \
+curl -X GET "https://api.callbell.eu/v1/contacts" \
```
Comment (proudlygeek) [inline docs/api_reference/contacts_api/get_contacts.md]: Why did you remove the extra params here? 🤔

File: docs/api_reference/subscriptions_api/_category_.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "label": "Subscriptions API",
```
Comment (proudlygeek) [inline docs/api_reference/subscriptions_api/_category_.json]: Maybe this should be documented inside the webhook section? Or at least referenced in the intro (e.g. there's also a programmatic way of doing this -> Link to this part)

File: docs/api_reference/subscriptions_api/subscribe.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+---
+title: POST /webhooks/subscribe
+sidebar_position: 4
+---
+
+# POST /contacts
+
+Creates a new webhook or updates an existing one. (Each account can have **only** one webhook at the time)
```
Comment (proudlygeek) [inline docs/api_reference/subscriptions_api/subscribe.md]: Maybe use a banner for this instead?

```suggestion
Creates a new webhook or updates an existing one.

:::warn
Each account can **only** register 1 webhook.
:::
```

**Review by proudlygeek — APPROVED** [review]
LGTM, I've suggested some changes around but LGTM 🙌

File: docs/api_reference/auth_api/introduction.md (hunk 1 above)
Comment (proudlygeek) [reply docs/api_reference/auth_api/introduction.md]:
```suggestion
Auth API allows to verify that the provided API token is valid and ready to be used.
```

File: docs/api_reference/contacts_api/get_contacts.md (hunk 2 above)
Comment (Qwertic) [reply docs/api_reference/contacts_api/get_contacts.md]: not sure actually, probably 🌮 🙏🏼

File: docs/api_reference/subscriptions_api/_category_.json (hunk 3 above)
Comment (Qwertic) [reply docs/api_reference/subscriptions_api/_category_.json]: Yes good idea, I can reference it in the webhooks guide too 👍🏼

File: docs/api_reference/subscriptions_api/subscribe.md (hunk 4 above)
Comment (Qwertic) [reply docs/api_reference/subscriptions_api/subscribe.md]: Sounds good 👍🏼

File: docs/api_reference/object_types/webhook.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+title: Webhook
+sidebar_position: 3
+---
+
+### Webhook Object
+
+| Field           | Type     | Description                                                                                                                                               |
+| :-------------- | :------- | :-------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `url`           | string   | A valid url where the events will be sent from Callbell                                                                                                   |
+| `subscriptions` | string[] | List of subcriptions containing subscribed events (`message_created`, `message_status_updated`, `contact_created`, `contact_updated`, `contact_deleted©`) |
```
Comment (proudlygeek) [inline docs/api_reference/object_types/webhook.md]:
```suggestion
| `subscriptions` | string[] | List of subcriptions containing subscribed events (`message_created`, `message_status_updated`, `contact_created`, `contact_updated`, `contact_deleted`) |
```

File: docs/api_reference/object_types/webhook_event.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+---
+
+### Webhook Event Object
+
+| Field       | Type     | Description                                                                                  |
+| :---------- | :------- | :------------------------------------------------------------------------------------------- |
+| `event`     | string   | Event type                                                                                   |
+| `response`  | string{} | Response body returned from external webhook endpoints                                       |
+| `payload`   | string{} | Payload object containing the subject of the event (message, contact)                        |
+| `success`   | booloean | Boolean value representing the status of the event request (`true` if status code was `200`) |
```
Comment (proudlygeek) [inline docs/api_reference/object_types/webhook_event.md]:
```suggestion
| `success`   | boolean | Boolean value representing the status of the event request (`true` if status code was `200`) |
```

File: docs/api_reference/webhooks/introduction.md [hunk 7]
```diff
@@ -12,6 +12,10 @@ Callbell API provides **HTTP Webhooks** for the most common application events:
 
 ## How to subscribe to events
 
+:::info
+This can be done also programmaticaly from [WebhooksEndpoint](./../webhooks_api/introduction.md)
```
Comment (proudlygeek) [inline docs/api_reference/webhooks/introduction.md]:
```suggestion
This can be done also programmatically from [WebhooksEndpoint](./../webhooks_api/introduction.md)
```

**Review by proudlygeek — APPROVED** [review]
LGTM 👍 I've added some minor corrections here and there

File: docs/api_reference/object_types/webhook.md (hunk 5 above)
Comment (Qwertic) [reply docs/api_reference/object_types/webhook.md]: 🌮

File: docs/api_reference/object_types/webhook_event.md (hunk 6 above)
Comment (Qwertic) [reply docs/api_reference/object_types/webhook_event.md]: 🌮

File: docs/api_reference/webhooks/introduction.md (hunk 7 above)
Comment (Qwertic) [reply docs/api_reference/webhooks/introduction.md]: 🌮
